src/main/java/org/cloudfoundry/promregator/auth/AuthenticationEnricherFactory.java

```diff
@@ -15,8 +15,13 @@ public static AuthenticationEnricher create(AuthenticatorConfiguration authConfi
 		AuthenticationEnricher ae = null;
 		
 		String type = authConfig.getType();
-		if ("OAuth2XSUAA".equalsIgnoreCase(type)) {
+		if("OAuth2XSUAA".equalsIgnoreCase(type)) {
+			log.warn("You are using deprecated authentication configuration type '{}'. Switch to '{}' instead.", type, "OAuth2XSUAABasic");
 			ae = new OAuth2XSUAAEnricher(authConfig.getOauth2xsuaa());
+		} else if ("OAuth2XSUAABasic".equalsIgnoreCase(type)) {
+			ae = new OAuth2XSUAAEnricher(authConfig.getOauth2xsuaaBasic());
+		} else if ("OAuth2XSUAACertificate".equalsIgnoreCase(authConfig.getType())) {
+			ae = new OAuth2XSUAAEnricher(authConfig.getOauth2xsuaaCertificate());
```

Comment:
I like this piece of logic a lot! Subtyping/polymorphism is like salt in a soup 😀